packages/api/amplify_api/lib/src/graphql/graphql_request_factory.dart

```diff
@@ -207,8 +207,10 @@ class GraphQLRequestFactory {
 
     // e.g. { 'name': { 'eq': 'foo }}
     if (queryPredicate is QueryPredicateOperation) {
+      final association = schema.fields?[queryPredicate.field]?.association;
+      // TODO(ragingsquirrel3): Change key logic when supporting CPK.
       final associatedTargetName =
-          schema.fields?[queryPredicate.field]?.association?.targetName;
+          association?.targetName ?? association?.targetNames?.first;
```

Comment:
mega nit: can be  `association?.targetNames?.first ?? association?.targetName` as the `targetNames` will be the default, `targetName` is the fallback for backwards compatibility.